src/components/elements/metadataTag/MetadataTag.tsx

```diff
@@ -0,0 +1,80 @@
+import { Box } from '@mui/system';
+import React from 'react';
+
+export function MetadataTag({
+  index,
+  label,
+  value,
+  maxWidth
+}: {
+  index: number;
+  label?: string;
+  value: any;
+  maxWidth?: number | string;
+}) {
+  const getColorByIndex = (index: number) => {
+    const regularColors = [
+      { backgroundColor: '#F3E8FF', color: '#7E22CE' },
+      { backgroundColor: '#EEF0FC', color: '#5467DE' },
+      { backgroundColor: '#CFFAFE', color: '#0E7490' },
+      { backgroundColor: '#DCFCE7', color: '#166534' },
+      { backgroundColor: '#FEF9C3', color: '#A16207' },
+      { backgroundColor: '#FFEDD5', color: '#C2410C' },
+      { backgroundColor: '#FEF2F2', color: '#DC2626' }
+    ];
+
+    const hoverColors = [
+      { backgroundColor: '#E9D5FF', color: '#7E22CE' },
+      { backgroundColor: '#CCD3FF', color: '#5467DE' },
+      { backgroundColor: '#A5F3FC', color: '#0E7490' },
+      { backgroundColor: '#BBF7D0', color: '#166534' },
+      { backgroundColor: '#FEF08A', color: '#A16207' },
+      { backgroundColor: '#FED7AA', color: '#C2410C' },
+      { backgroundColor: '#FEE2E2', color: '#DC2626' }
+    ];
+
+    const colorIndex = index % 7;
+    return {
+      regular: regularColors[colorIndex],
+      hover: hoverColors[colorIndex]
+    };
+  };
```

Comment:
```sugestion
    const generateColors = (hue: number) => {
        const saturation = 100;
        const lightness = 95;
        const hoverLightness = 92;
        const textColorLightness = 35;

        const bg = `hsl(${hue}, ${saturation}%, ${lightness}%)`;
        const hoverBg = `hsl(${hue}, ${saturation}%, ${hoverLightness}%)`;
        const text = `hsl(${hue}, ${saturation-40}%, ${textColorLightness}%)`;

        return {
            bg,
            hoverBg,
            text
        }
    }

    const getColorByIndex = (index: number) => {
    const {bg, hoverBg, text} = generateColors(index * 300);

    return {
        regular: { backgroundColor: bg, color: text },
        hover: { backgroundColor: hoverBg, color: text }
    };
```
Seems to generate colors pretty similar to the ones we use.
Also, I think the component should receive the `colorPalette` as a prop, and the color choice should be done outside.